src/Illuminate/Routing/Router.php

```diff
@@ -630,6 +630,11 @@ protected function runRoute(Request $request, Route $route)
             return $route;
         });
 
+        $routeExceptionHandler = $route->getHandler();
+        if (! is_null($routeExceptionHandler)) {
+            app()->singleton('Illuminate\Contracts\Debug\ExceptionHandler', $routeExceptionHandler);
```

Comment:
There is the container instance in this class, so no need to use the `app` helper :)